android/src/com/mapswithme/maps/widget/placepage/PlacePageButtons.java

```diff
@@ -308,22 +310,24 @@ private void preserveRoutingButtons(@NonNull List<PlacePageButton> items, @NonNu
     }
   }
 
-  private ArrayList<MenuBottomSheetItem> getMenuItems(final List<PlacePageButton> buttons)
+  public ArrayList<MenuBottomSheetItem> getMenuBottomSheetItems()
   {
+    if (mMoreItems == null)
+      return null;
     ArrayList<MenuBottomSheetItem> items = new ArrayList<>();
-    for (int i = mMaxButtons; i < buttons.size(); i++)
+    for (int i = mMaxButtons; i < mMoreItems.size(); i++)
```

Comment:
Passing buttons as an argument had a more local scope and fewer potential side effects.

Reply:
Yes but passing the buttons as argument means losing the state on config change.

The bottom sheet fragment will be recreated without the host activity calling the `show` method, thus the activity will not be able to pass the buttons.

The way it is now implemented, the bottom sheet fragment will be the one calling this function using the activity context. I don't really like this approach but I could not find  cleaner way to do this.

I could make the host activity/fragment retrieve a reference to the bottom sheet using its tag, but I don't know if it will be cleaner when there are multiple bottom sheets in one activity/fragment.